### ISSUE TITLE
HATS transformation generates a different number of cells per sample during batching

### ISSUE THREAD

**Issue by fabhertz95** [neuromorphs/tonic#180]
Hi everyone! I am experience a problem when trying to train an SVM on batches generated through PyTorch and Tonic. 
In particular, I am using the NMNIST dataset from Tonic, to which a transform is applied to get the HATS representation of each events stream. The code is the following:

```
import tonic
import tonic.transforms as transforms
import torch
from torch.utils.data import DataLoader

# Loading data in batch mode.                                                                                                                                     
batch_size=16
loader = DataLoader(dataset_tr, batch_size=batch_size, shuffle=True, collate_fn=tonic.collation.PadTensors(batch_first=True))
for h, y in loader:
  print(h.shape)
```
If one runs this code, it can be noticed how the number of cells (second dimension in the histogram shape) increases after each batch iteration. How is this possible since the number of cells is predetermined by the surface extension and the sensor size?
In my implementation of HATS I have not this problem; however, I have not been able to replicate the HATS paper results (I get a classification accuracy of 98.77% on NMNIST, while the authors get 99.2% with a linear SVM). For this reason, I wanted to compare my HATS implementation with yours. 

Thank you in advance for your help! And thank you for your great work!

**Comment by fabhertz95** [comment]
> 

I forgot a piece of code which has to be put on top :) 

```
HATS_transform = transforms.ToAveragedTimesurface(sensor_size=(32,32,2), cell_size=5, surface_size=5, temporal_window=1e11, tau=1e8, decay='exp')
dataset_tr = tonic.datasets.NMNIST(save_to="./train_data", train=True, transform=HATS_transform)
```

**Comment by biphasic** [comment]
Hey @fabhertz95 
really sorry for the late answer. The HATS representation is a community contribution and to be honest is likely to be flawed. I remember that some people in my old lab also tried to replicate the results of that paper and they couldn't do it either. They tried to contact the authors but didn't get any help. 

Regarding your implementation, do you think you would be willing to contribute your version to Tonic? We can also have a chat on Discord if you like, it will be easier to discuss there https://discord.gg/V6FHBZURkg

**Comment by biphasic** [comment]
merged your PR #184
